src/routes/booking.route.js

```diff
@@ -3,14 +3,24 @@ import BookingsController from '../controllers/bookings.controller';
 import BookingMiddleware from '../middlewares/booking.middleware';
 import UserMiddleware from '../middlewares/userMiddlware';
 import validation from '../validation';
+import AccommodationMiddleware from '../middlewares/accommodation.middleware';
+import RequestMiddleware from '../middlewares/requestMiddlware';
 
 const app = express.Router();
 
 const { checkToken } = UserMiddleware;
-const { checkUserBooking } = BookingMiddleware;
-const { setAccommodationRating } = BookingsController;
-const { reviewDateValidation, reviewValidation } = validation;
+const { checkUserBooking, bookingExist } = BookingMiddleware;
+const { setAccommodationRating, createBooking, getBooking } = BookingsController;
+const { reviewDateValidation, reviewValidation, validateBooking } = validation;
+const { checkIfTripExists, checkTripOwner } = RequestMiddleware;
+const { accommodationExists, isAccommodationAvailable } = AccommodationMiddleware;
 
 app.patch('/:id/rate', reviewValidation, checkToken, checkUserBooking, reviewDateValidation, setAccommodationRating);
+app.post(
+  '/', checkToken, validateBooking,
+  accommodationExists, isAccommodationAvailable, checkIfTripExists, checkTripOwner,
+  createBooking
+);
+app.get('/:id', checkToken, bookingExist, checkUserBooking, getBooking);
```

Comment:
awesome work! However, there is no validation for this id, it  crashing when a letter is provided as id